python/arcticdb/version_store/library.py

```diff
@@ -105,7 +106,7 @@ class NameWithDType(NamedTuple):
     """A named tuple. A name and dtype description pair."""
 
     name: str
```

Comment:
For the `index` property we might not have a name, in which case `None` is returned. See:

```python
>>> l.get_description('s').columns[0].name
'a'
>>> l.get_description('s').index.name
[None]
```

So I'd suggest we make this change:

```suggestion
    name: Optional[str]
```